lang/cs/Org.Apache.REEF.Client/YARN/YarnREEFDotNetClient.cs

```diff
@@ -0,0 +1,216 @@
+            string localDriverFolderPath = CreateDriverFolder(jobId, applicationId);
+            Log.Log(Level.Verbose, "Preparing driver folder in {0}", localDriverFolderPath);
+            _driverFolderPreparationHelper.PrepareDriverFolder(jobSubmission, localDriverFolderPath);
+
+            // prepare configuration
+            var paramInjector = TangFactory.GetTang().NewInjector(jobSubmission.DriverConfigurations.ToArray());
+            int maxApplicationSubmissions =
+                paramInjector.GetNamedInstance<DriverBridgeConfigurationOptions.MaxApplicationSubmissions, int>();
+
+            var avroJobSubmissionParameters = new AvroJobSubmissionParameters
+            {
+                jobId = jobSubmission.JobIdentifier,
+                tcpBeginPort = paramInjector.GetNamedInstance<TcpPortRangeStart, int>(),
+                tcpRangeCount = paramInjector.GetNamedInstance<TcpPortRangeCount, int>(),
+                tcpTryCount = paramInjector.GetNamedInstance<TcpPortRangeTryCount, int>(),
+                jobSubmissionFolder = localDriverFolderPath
+            };
+
+            var avroYarnJobSubmissionParameters = new AvroYarnJobSubmissionParameters
+            {
+                driverMemory = jobSubmission.DriverMemory,
+                driverRecoveryTimeout = paramInjector.GetNamedInstance<DriverBridgeConfigurationOptions.DriverRestartEvaluatorRecoverySeconds, int>(),
+                jobSubmissionDirectoryPrefix = jobSubmissionDirectory,
+                dfsJobSubmissionFolder = jobSubmissionDirectory,
+                sharedJobSubmissionParameters = avroJobSubmissionParameters
+            };
+
+            var submissionArgsFilePath = Path.Combine(localDriverFolderPath,
+                _fileNames.GetLocalFolderPath(),
+                _fileNames.GetJobSubmissionParametersFile());
+            using (var argsFileStream = new FileStream(submissionArgsFilePath, FileMode.CreateNew))
+            {
+                var serializedArgs = AvroJsonSerializer<AvroYarnJobSubmissionParameters>.ToBytes(avroYarnJobSubmissionParameters);
+                argsFileStream.Write(serializedArgs, 0, serializedArgs.Length);
+            }
+
+            // upload prepared folder to DFS
+            var jobResource = _jobResourceUploader.UploadJobResource(localDriverFolderPath, jobSubmissionDirectory);
+
+            // submit job
+            Log.Log(Level.Verbose, @"Assigned application id {0}", applicationId);
+
+            var submissionReq = CreateApplicationSubmissionRequest(jobSubmission,
+                applicationId,
+                maxApplicationSubmissions,
+                jobResource);
+            var submittedApplication = _yarnRMClient.SubmitApplicationAsync(submissionReq).GetAwaiter().GetResult();
+            Log.Log(Level.Info, @"Submitted application {0}", submittedApplication.Id);
+        }
+
+        public IJobSubmissionResult SubmitAndGetJobStatus(IJobSubmission jobSubmission)
```

Comment:
Will this be implemented? If not, maybe `NotSupportedException` is a better choice.